src/ert/gui/suggestor/_suggestor_message.py

```diff
@@ -58,39 +55,121 @@ def __init__(
+            self._expand_collapse_label.setOpenExternalLinks(False)
+            self._expand_collapse_label.linkActivated.connect(self._toggle_expand)
+
+            self._vbox = QWidget()
+            layout = QVBoxLayout()
+            self._vbox.setLayout(layout)
+            layout.addWidget(self.lbl)
+            layout.addWidget(self._expand_collapse_label)
+            self._hbox.addWidget(self._vbox, alignment=Qt.AlignmentFlag.AlignTop)
+        else:
+            self._expand_collapse_label = QLabel()
+            self._hbox.addWidget(self.lbl, alignment=Qt.AlignmentFlag.AlignTop)
+
+    def _collapsed_text(self) -> str:
+        location_paragraph = ""
+        if self._locations:
+            location_paragraph = self._locations[0]
+            location_paragraph = (
+                "<p>"
+                + f'<b style="color: {self._text_color}"> location: </b>'
+                + location_paragraph
+                + "</p>"
+            )
+
+        return (
             '<div style="font-size: 16px; line-height: 24px;">'
-            + f'<b style="color: {text_color}">'
-            + header
+            + f'<b style="color: {self._text_color}">'
+            + self._header
             + "</b>"
-            + info.message
-            + "<p>"
-            + info.location()
-            + "</p>"
+            + self._message
+            + location_paragraph
             + "</div>"
         )
-        self.lbl.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)
-        self.lbl.setTextInteractionFlags(Qt.TextInteractionFlag.TextSelectableByMouse)
-        self.lbl.setWordWrap(True)
 
-        self.hbox = QHBoxLayout()
-        self.hbox.setContentsMargins(16, 16, 16, 16)
-        self.hbox.addWidget(self.icon, alignment=Qt.AlignmentFlag.AlignTop)
-        self.hbox.addWidget(self.lbl, alignment=Qt.AlignmentFlag.AlignTop)
-        self.setLayout(self.hbox)
+    def _toggle_expand(self, _link: Any) -> None:
+        if self._expanded:
+            self.lbl.setText(self._collapsed_text())
+            self._expand_collapse_label.setText(self._expand_link())
+            self._expanded = False
+        else:
+            self.lbl.setText(self._expanded_text())
+            self._expand_collapse_label.setText(self._hide_link())
+            self._expanded = True
+
+    def _hide_link(self) -> str:
+        return " <a href=#morelocations>show less</a>"
+
+    def _expand_link(self) -> str:
+        return f" <a href=#morelocations>and {len(self._locations) - 1} more</a>"
+
+    def _expanded_text(self) -> str:
+        location_paragraphs = ""
+        first = True
+        for loc in self._locations:
+            if first:
+                location_paragraphs += (
+                    f'<p><b style="color: {self._text_color}">location:</b>{loc}</p>'
+                )
+                first = False
+            else:
+                location_paragraphs += f"<p>{loc}</p>"
```

Comment:
This is also be a candidate to extract as a separate formatting function?
I am unsure if the readability is better after that though, but at least we avoid differing implementations if that would ever happen.

Reply:
Not sure if I extracted the duplication that you wanted but did extract some.